packages/uploadthing/src/sdk/index.ts

```diff
@@ -14,6 +16,61 @@ function getApiKeyOrThrow() {
   return process.env.UPLOADTHING_SECRET;
 }
 
+// File is just a Blob with a name property
+type FileEsque = Blob & { name: string };
+
+/**
+ * @param {FileEsque | FileEsque[]} files The file(s) to upload
+ * @param {Json} metadata JSON-parseable metadata to attach to the uploaded file(s)
+ */
+export const uploadFiles = async (
+  files: FileEsque[] | FileEsque,
+  metadata: Json = {},
+) => {
+  guardServerOnly();
+
+  if (!Array.isArray(files)) files = [files];
+
+  const formData = new FormData();
+  files.forEach((file) => formData.append("files", file));
+  formData.append("metadata", JSON.stringify(metadata));
+
+  return uploadFilesInternal(formData, {
+    apiKey: getApiKeyOrThrow(),
+    utVersion: UPLOADTHING_VERSION,
+  });
+};
+
+/**
+ * @param {string} url The URL of the file to upload
+ * @param {Json} metadata JSON-parseable metadata to attach to the uploaded file(s)
+ */
+export const uploadFileFromUrl = async (
+  url: string | URL,
+  metadata: Json = {},
+) => {
+  guardServerOnly();
+
+  if (typeof url === "string") url = new URL(url);
+  const filename = url.pathname.split("/").pop() ?? "unknown-filename";
+
+  // Download the file on the user's server to avoid egress charges
+  const fileResponse = await fetch(url);
+  if (!fileResponse.ok) {
+    throw new Error("Failed to download file");
+  }
+  const blob = await fileResponse.blob();
+
+  const formData = new FormData();
+  formData.append("files", blob, filename);
+  formData.append("metadata", JSON.stringify(metadata));
+
+  return uploadFilesInternal(formData, {
+    apiKey: getApiKeyOrThrow(),
+    utVersion: UPLOADTHING_VERSION,
+  }).then((files) => files[0]);
```

Comment:
Should this function also take an array or urls?

Reply:
It can, just didnt think it was really a use case here. For direct file uploads it makes sense since you can selektiv multiple files but I didnt think the same really applied for URLs.

But i can definetely add it. Should the return type be an array if and only if an array is passed in?

Reply:
I can't think of a specific use case either, it was just thinking so that the API was similar for both uploadFiles and uploadFilesViaUrl.

> Should the return type be an array if and only if an array is passed in?

This makes sense to me

Reply:
Done